### ISSUE TITLE
Expose an value's tag on YAML::Any

### ISSUE THREAD

**Issue by obskyr** [crystal-lang/crystal#10646]
Currently (as of Crystal 1.0.0), Crystal's standard YAML library doesn't expose access to tags. It would be helpful to be able to access these – and similarly, it would be preferable if `YAML.dump(YAML.parse(string))` didn't strip tags. (My personal use case for it is that I'm designing a small YAML-based DSL, and I'd like to use tags to signify macros!)

As in #3516, (which was for some reason closed by the author), I propose exposing `#tag` on `YAML::Any`, simply letting the user do, for example:

```crystal
require "yaml"

yaml = "---
- !!str
  123
- !custom
  def
- 456
"

parsed = YAML.parse yaml
yaml[0].tag # => "tag:yaml.org,2002:str"
yaml[1].tag # => "custom"
yaml[2].tag # => "tag:yaml.org,2002:int"
```

Separate from #8598, this is a bit more of a lightweight feature – letting the user access the tags after the fact and falling back on YAML's automatic tag resolution (as Crystal's parser already does), rather than letting the user specify special parsers.

One piece of the YAML spec that might help especially in the implementation of this is [3.3.2. Resolved Tags](https://yaml.org/spec/1.2/spec.html#id2768011)!

**Comment by asterite** [comment]
Note that you can do `YAML::Nodes.parse(...)` and it will give a YAML structure where you can inspect tags, the quote style, etc.

**Comment by asterite** [comment]
That is, YAML::Any is inteded to be used when you don't care about tags. If you do care about such information it means you need a bit more low-level info, and then you need to use a bit more low-level API, which is `YAML::Nodes.parse(...)`.

**Comment by straight-shoota** [comment]
I agree. `YAML::Any` is a basic mapping of YAML data structures to Crystal data types based on the YAML Core schema.

`YAML::Nodes` is the appropriate data structure for representing a YAML document including specific like tags.
